packages/cells/src/widget.ts

```diff
@@ -1603,6 +1603,11 @@ export class CodeCell extends Cell<ICodeCellModel> {
     }
     // This is to hide/show icon on single line output.
     this.updatePromptOverlayIcon();
+
+    // Clear output area when empty
+    if (this.model.outputs.length === 0) {
+      this.outputArea.node.style.height = '';
+    }
```

Comment:
Thank you for picking this up @NexVeridian, it looks good! Here is a way to improve it before we merge it: users expect that when they re-run a single cell it will retain the height they set on the output. However, as cell gets re-run it gets back to 0 outputs for a brief moment so the height would be lost due to this conditional.

Could you store the value of `this.outputArea.node.style.height` in a private variable before clearing it and then if the number of outputs is greater than 0 restore it?